source/tutorials/kubernetes.rst

```diff
@@ -3,170 +3,340 @@
+
+   --pod-network-cidr 10.244.0.0/16
 
-         sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps) 2>/dev/null
-         sudo swapoff -a
 
-      .. note::
 
-         On systems with limited resources, some performance degradation may
-         be observed while swap is disabled.
+Choose a container runtime
+==========================
 
-#. Switch to root to modify the `hosts` file:
+Container runtimes are the underlying fabric that pod workloads execute inside
+of. Different container runtimes offer different balances between features,
+performance, and security. 
+
+Kubernetes allows integration various container runtimes via a container
+runtime interface (CRI). |CL| supports Kubernetes with the various runtimes
+below:
+
+* `Docker (Dockershim)`_ with or without `Kata Containers`_.
+* `containerd`_ with or without `Kata Containers`_.
+* `CRI+O`_ with or without `Kata Containers`_.
+
+
+The container runtime that you choose will dictate the steps necessary to
+initialize the master cluster with :command:`kubeadm init`.
+
+
+Docker (Dockershim)
+-------------------
+
+The default built-in runtime provided by Kubernetes is using the system Docker
+via Dockershim and as a result is one of the simplest to use. 
+
+One limitation of using Dockershim is that all pods on the Kubernetes node
+will inherit and use the default runtime that Docker is set to use. To be able
+to specify a container runtime per-Kerbernetes service, use CRI+O or
+containerd. 
+
+
+#. Make sure Docker is installed:
+
+   .. code:: bash
+
+      sudo swupd bundle-add containers-basic
+
+#. Start the Docker service and enable it to start automatically at boot:
+
+   .. code::
+
+      sudo systemctl enable --now docker.service
+
+
+#. Run the kubeadm command to initialize the master node:
+
+   .. important:: 
+
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. In this example, *flannel* is
+      being used for the pod networking. See `Choose a pod network add-on`_
+      for more information.
+
+   .. code:: bash
+
+      sudo kubeadm init \
+      --pod-network-cidr 10.244.0.0/16 #required for flannel
+
+
+.. warning:: 
+
+   Docker on |CL| will automatically use kata-runtime as the default Docker
+   runtime if it is available.
+
+   If you do not want to use the kata runtime or experience problems with
+   cluster initialization, you can disable Docker from setting kata as the
+   default runtime by running these commands:
+
+   .. code:: bash 
+
+      sudo rm /etc/systemd/system/docker.service.d/50-runtime.conf
+      sudo systemctl mask docker-set-default-runtime.service
+
+#. Once the cluster initialization is complete, continue reading about how to
+   `Use your cluster`_.
+
+containerd
+----------
+
+`containerd <https://containerd.io/>`_ is the runtime  that the Docker engine
+is built ontop of. Kubernetes can use containerd directly instead of going
```

Comment:
"on top"

Reply:
Thanks. Since this is editorial, I'll prefer TCS to do this in a follow up PR.